apm-agent-core/pom.xml

```diff
@@ -165,6 +165,14 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>9</source>
```

Comment:
Why is Java 9 required here? The agent must still support Java 7.  Loading the plugin (instantiating `FutureInstrumentation`) on an earlier JVM would result in an error.

Reply:
I'll have to clean up the PR a bit later on, some parts were needed to get it running on my laptop. Otherwise, it was complaining about the java versions.